modules/ca/src/tools/caget.c

```diff
@@ -389,9 +391,21 @@ int main (int argc, char *argv[])
 
     int nPvs;                   /* Number of PVs */
     pv* pvs;                    /* Array of PV structures */
+    const char* tmo;            /* timeout from environment var */
 
     LINE_BUFFER(stdout);        /* Configure stdout buffering */
 
+    if ((tmo = getenv(DEFAULT_TIMEOUT_ENV)) != NULL)
+    {
+    if(epicsScanDouble(tmo, &caTimeout) != 1)
+            {
+                fprintf(stderr, "'%s' is not a valid timeout value "
+                        "(from '%s' in the environment) - ignored. ('caget -h' for help.)\n", tmo, DEFAULT_TIMEOUT_ENV);
+                caTimeout = DEFAULT_TIMEOUT;
+            }
+
+    }
+
```

Comment:
Yank this into a function in tool_lib.c

Reply:
But be careful about the tool name in the error message...

Reply:
code was moved as suggested.